pkg/api/routes.go

```diff
@@ -1531,7 +1531,7 @@ func (rh *RouteHandler) ListRepositories(response http.ResponseWriter, request *
 		return
 	}
 
-	if acCtx != nil {
+	if acCtx != nil && len(combineRepoList) != 0 {
```

Comment:
Why do we need this change? Theoretically we should be able to iterate over an empty slice.

Reply:
because a sync test was failing, but I guess we can modify the test.

the test expected []string{} but now it gets []string(nil) because now acCtx will not be nil, being set in authn.

let's make sure we don't return {"repositories": [nil]} in /v2/_catalog response when there is no repos.

Reply:
Why would we return `[nil]`? I guess that's the bug.

Reply:
Before setting acCtx in authn.go, it entered the else branch, where the repo list was set on nil, now because acCtx isn't nil anymore here, it would enter the if branch and remain a list

Reply:
We need to clarify this in a call.